### PR TITLE
fix: remove extra padding from button with icon only

### PR DESCRIPTION
**Pull request by fredmaggiowski**
### Description

Remove extra padding from buttons with only icon.

I've also updated antd to v5.17 to leverage the new `Button#iconPosition` prop and remove our custom code

### Checklist

- [x] commit message and branch name follow conventions
- [x] tests are included
- [x] changes are accessible and documented from components stories
- [x] typings are updated or integrated accordingly with your changes
- [x] all added components are exported from index file (if necessary)
- [x] all added files include Apache 2.0 license
- [x] you are not committing extraneous files or sensitive data
- [x] the browser console does not have any logged errors
- [x] necessary labels have been applied to this pull request (enhancement, bug, ecc.)

### CODE REVIEW

File: src/components/Button/Button.module.css [hunk 1]
```diff
@@ -22,6 +22,11 @@
   justify-content: center;
   gap: var(--spacing-gap-xs, 4px);
   box-shadow: none;
+
+  & :global(.mia-platform-btn-icon) {
```
Comment (Danielecina) [inline src/components/Button/Button.module.css:26]: Can you add a comment explaining this requirement?

File: src/components/Button/__snapshots__/Button.test.tsx.snap [hunk 2]
```diff
@@ -38,23 +38,27 @@ exports[`Button Component renders button with icon left correctly 1`] = `
...
-      fill="currentColor"
-      height="16"
-      role="img"
-      stroke="currentColor"
-      stroke-width="0"
-      style="color: white;"
-      viewBox="0 0 256 256"
-      width="16"
-      xmlns="http://www.w3.org/2000/svg"
+    <span
```
Comment (Danielecina) [inline src/components/Button/__snapshots__/Button.test.tsx.snap:41]: why do we have a new wrapper?

File: src/components/Button/Button.module.css (hunk 1 above)
Comment (fredmaggiowski) [reply src/components/Button/Button.module.css:26]: pretty straightforward: antd sets a padding based on the iconPosition but our button is already setting a gap

File: src/components/Button/__snapshots__/Button.test.tsx.snap (hunk 2 above)
Comment (fredmaggiowski) [reply src/components/Button/__snapshots__/Button.test.tsx.snap:41]: made by antd, i don't know tbh

**Review by Danielecina — APPROVED** [review]

**Pull request merged**